### ISSUE TITLE
Training speed is not changed by batch size in 0.6.4 version.

### ISSUE THREAD

**Issue by marunguy** [plaidml/plaidml#409]
* python 3.6.8 64bit, windows 10 64bit, rtx-2080ti

Training speed is not changed by batch size in 0.6.4 version.

---
* plaidml 0.6.4, batch_size=32
'''
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
4096/4096 [==============================] - 9s 2ms/step - loss: 2.3064
Epoch 2/10
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3030
Epoch 3/10
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3029
'''
* plaidml 0.6.4, batch_size=64
'''
plaidml_keras36) d:\temp>python keras_vgglikeconvnet.py
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
4096/4096 [==============================] - 10s 2ms/step - loss: 2.3085
Epoch 2/10
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3028
Epoch 3/10
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3031
'''
* plaidml 0.6.4, batch_size=128
'''
(plaidml_keras36) d:\temp>python keras_vgglikeconvnet.py
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
4096/4096 [==============================] - 6s 2ms/step - loss: 2.3080
Epoch 2/10
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3020
Epoch 3/10
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3020
'''
* plaidml 0.6.4, batch_size=256
'''
(plaidml_keras36) d:\temp>python keras_vgglikeconvnet.py
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
4096/4096 [==============================] - 8s 2ms/step - loss: 2.3136
Epoch 2/10
4096/4096 [==============================] - 5s 1ms/step - loss: 2.3067
Epoch 3/10
4096/4096 [==============================] - 5s 1ms/step - loss: 2.3038
'''
* plaidml 0.6.4, batch_size=512
'''
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
4096/4096 [==============================] - 6s 2ms/step - loss: 2.3143
Epoch 2/10
4096/4096 [==============================] - 5s 1ms/step - loss: 2.3052
Epoch 3/10
4096/4096 [==============================] - 5s 1ms/step - loss: 2.3038
'''
---
* plaidml 0.6.3, batch_size=32
'''
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
INFO:plaidml:Analyzing Ops: 121 of 239 operations complete
4096/4096 [==============================] - 6s 2ms/step - loss: 2.3078
Epoch 2/10
4096/4096 [==============================] - 4s 896us/step - loss: 2.3032
Epoch 3/10
4096/4096 [==============================] - 4s 891us/step - loss: 2.3030
'''
* plaidml 0.6.3, batch_size=64
'''
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
INFO:plaidml:Analyzing Ops: 121 of 239 operations complete
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3092
Epoch 2/10
4096/4096 [==============================] - 3s 781us/step - loss: 2.3031
Epoch 3/10
4096/4096 [==============================] - 3s 779us/step - loss: 2.3030
'''
* plaidml 0.6.3, batch_size=128
'''
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
INFO:plaidml:Analyzing Ops: 121 of 239 operations complete
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3125
Epoch 2/10
4096/4096 [==============================] - 3s 753us/step - loss: 2.3037
Epoch 3/10
4096/4096 [==============================] - 3s 756us/step - loss: 2.3025
'''
* plaidml 0.6.3, batch_size=256
'''
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
INFO:plaidml:Analyzing Ops: 127 of 239 operations complete
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3105
Epoch 2/10
4096/4096 [==============================] - 3s 732us/step - loss: 2.3038
Epoch 3/10
4096/4096 [==============================] - 3s 732us/step - loss: 2.3026

* plaidml 0.6.3, batch_size=512
'''
INFO:plaidml:Opening device "opencl_nvidia_geforce_rtx_2080_ti.0"
Epoch 1/10
INFO:plaidml:Analyzing Ops: 127 of 239 operations complete
4096/4096 [==============================] - 6s 1ms/step - loss: 2.3172
Epoch 2/10
4096/4096 [==============================] - 3s 706us/step - loss: 2.3049
Epoch 3/10
4096/4096 [==============================] - 3s 706us/step - loss: 2.3033
'''
* test code
VGG-like convnet from https://keras.io/getting-started/sequential-model-guide/ 
The number of data(100->4096) and batch_size(32->128) are modified.
```python
try:
    import plaidml.keras
except ImportError:
    pass
else:
    plaidml.keras.install_backend()

import numpy as np
import keras
from keras.models import Sequential
from keras.layers import Dense, Dropout, Flatten
from keras.layers import Conv2D, MaxPooling2D
from keras.optimizers import SGD

# Generate dummy data
x_train = np.random.random((4096, 100, 100, 3))
y_train = keras.utils.to_categorical(np.random.randint(10, size=(4096, 1)), num_classes=10)
x_test = np.random.random((20, 100, 100, 3))
y_test = keras.utils.to_categorical(np.random.randint(10, size=(20, 1)), num_classes=10)

model = Sequential()
# input: 128x128 images with 3 channels -> (100, 100, 3) tensors.
# this applies 32 convolution filters of size 3x3 each.
model.add(Conv2D(32, (3, 3), activation='relu', input_shape=(100, 100, 3)))
model.add(Conv2D(32, (3, 3), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Conv2D(64, (3, 3), activation='relu'))
model.add(Conv2D(64, (3, 3), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(256, activation='relu'))
model.add(Dropout(0.5))
model.add(Dense(10, activation='softmax'))

sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
model.compile(loss='categorical_crossentropy', optimizer=sgd)

model.fit(x_train, y_train, batch_size=128, epochs=10)
score = model.evaluate(x_test, y_test, batch_size=32)
```
